base/random.jl

```diff
@@ -683,6 +685,39 @@ end
 rand{T}(rng::AbstractRNG, r::AbstractArray{T}, dims::Dims) = rand!(rng, Array{T}(dims), r)
 rand(rng::AbstractRNG, r::AbstractArray, dims::Int...) = rand(rng, r, dims)
 
+# rand from a string
+
+isvalid_unsafe(s::String, i) = !Base.is_valid_continuation(unsafe_load(pointer(s), i))
+
+function rand(rng::AbstractRNG, s::String)::Char
+    g = RangeGenerator(1:s.len)
+    while true
+        pos = rand(rng, g)
+        isvalid_unsafe(s, pos) && return s[pos]
+    end
+end
+
+function rand(rng::AbstractRNG, s::AbstractString)::Char
+    g = RangeGenerator(1:endof(s))
+    while true
+        try # the generic `isvalid` includes an equivalent try/catch statement
+            return s[rand(rng, g)]
+        end
+    end
+end
```

Comment:
@StefanKarpinski would you mind having a look at this updated generic implementation, to confirm it complies with `AbstractString` API?